support-modules/supporter-product-data-dynamo/src/main/scala/com/gu/supporterdata/services/SupporterDataDynamoService.scala

```diff
@@ -32,33 +32,49 @@ class SupporterDataDynamoService(client: DynamoDbAsyncClient, tableName: String)
     // Dynamo will delete expired subs at the start of the day, whereas the subscription actually lasts until the end of the day
     val expiryDate = item.termEndDate.plusDays(1)
     val expiryDateName = "expiryDate"
+    val contributionAmount = "contributionAmount"
+    val contributionCurrency = "contributionCurrency"
 
     val key = Map(
       identityId -> AttributeValue.builder.s(beneficiaryIdentityId).build,
       subscriptionName -> AttributeValue.builder.s(item.subscriptionName).build,
     ).asJava
 
-    val updateExpression =
+    val nonOptionExpression =
```

Comment:
I've split the update expression out into optional and non optional elements as contribution amount and currency are only needed for contributions